compiler/noirc_evaluator/src/ssa/opt/pure.rs

```diff
@@ -28,20 +27,15 @@ impl Ssa {
     /// identified as calling known pure functions.
     #[tracing::instrument(level = "trace", skip(self))]
     pub(crate) fn purity_analysis(mut self) -> Ssa {
-        let mut purities = HashMap::default();
-        let mut called_functions = HashMap::default();
-
         // First look through each function to get a baseline on its purity and collect
         // the functions it calls to build a call graph.
-        for function in self.functions.values() {
-            let (purity, dependencies) = function.is_pure();
-            purities.insert(function.id(), purity);
-            called_functions.insert(function.id(), dependencies);
-        }
+        let purities: HashMap<_, _> =
+            self.functions.values().map(|function| (function.id(), function.is_pure())).collect();
```

Comment:
Is the comment above this still accurate? Looks like the call dependencies are moved below